dessia_common/workflow/core.py

```diff
@@ -1941,6 +1943,138 @@ def _method_jsonschemas(self):
+        """ Computes the input values for blocks and non-block variables. """
+        values_nonblock_variables = []
+        values_blocks = []
+        for i, variable in enumerate(self.workflow.inputs):
+            if variable in self.workflow.nonblock_variables:
+                values_nonblock_variables.append(self.input_values[i])
+            else:
+                values_blocks.append(self.input_values[i])
+
+        return values_nonblock_variables, values_blocks
+
+    def _to_script(self, workflow_script):
+        """ Computes elements for a to_script interpretation. """
+        input_info = {
+            "input_str": "",
+            "default_value": "",
+            "add_import": [],
+        }
+
+        block_input = "block_{}.inputs[{}]"
+
+        values_nonblock_variables, values_blocks = self._compute_input_values()
+
+        # Blocks
+        index_ = 0
+        for j, block in enumerate(self.workflow.blocks):
+            for i, input_ in enumerate(block.inputs):
+                input_key = block_input.format(j, i)
+                if input_key not in workflow_script.declaration:
+                    input_info["input_str"] += f"    workflow.input_index({input_key}): value_{j}_{i},\n"
+                    input_info["default_value"] += self._generate_default_value(values_blocks[index_], j, i)
+                    index_ += 1
+                    input_info["add_import"].extend(get_schema(annotation=input_.type_,
+                                                               attribute=SchemaAttribute(input_.name)).get_import_names(
+                        import_names=[]))
+
+        # NBVs
+        for k, nbv in enumerate(self.workflow.nonblock_variables):
+            input_info["input_str"] += f"    workflow.input_index(variable_{k}): value_{k}_,\n"
+            input_info["default_value"] += self._generate_default_value(values_nonblock_variables[k], k)
+            input_info["add_import"].extend(get_schema(annotation=nbv.type_,
+                                                       attribute=SchemaAttribute(nbv.name)).get_import_names(
+                import_names=[]))
+
+        return input_info["input_str"], input_info["default_value"], input_info["add_import"]
+
+    def to_script(self):
+        """ Computes a script representing the WorkflowRun. """
+        workflow_script = self.workflow._to_script()
+        input_str, default_value, add_import = self._to_script(workflow_script)
+
+        workflow_script.declaration = f"{workflow_script.declaration}\n{default_value}\n\ninput_values =" \
+                                      f" {{\n{input_str}}}\nworkflow_run = workflow.run(input_values=input_values)\n"
+        workflow_script.imports.extend(add_import)
+
+        return workflow_script
+
+    def _generate_default_value(self, value, block_index, input_index=None):
+        """ Generate a value for a given input. """
+        if not input_index and input_index != 0:
```

Comment:
What is the usecase of this ? Why not just `if input_index is None` ?

Reply:
because if input_index is zero it is considered None, and if it's equal to zero I need it

Reply:
done !